lib/cloud_controller/diego/docker/staging_completion_handler.rb

```diff
@@ -36,6 +37,12 @@ def save_staging_result(app, payload)
               end
             end
 
+            if payload.key?(:lifecycle_data) && payload[:lifecycle_data][:docker_image].present?
+              droplet.update_cached_docker_image(payload[:lifecycle_data][:docker_image])
+            else
+              droplet.update_cached_docker_image(nil)
+            end
+
```

Comment:
The idea here was to remove the `cached_docker_image` if the user opted-out of the docker caching. This should allow re-staging with another docker image ref/url and starting new instances of the app.